robot/Cumulus/resources/RecurringDonationsPageObject.py

```diff
@@ -7,30 +7,60 @@
 from datetime import datetime
 from dateutil.relativedelta import relativedelta
 
+
 @pageobject("Listing", "npe03__Recurring_Donation__c")
 class RDListingPage(BaseNPSPPage, ListingPage):
     object_name = "npe03__Recurring_Donation__c"
 
+    @capture_screenshot_on_error
+    def click_rd2_modal_button(self, name):
+      """Based on the button name (Cancel)  or (Save) on the modal footer, select and click on the respective button"""
+      btnlocator = npsp_lex_locators["button-with-text"].format(name)
+      self.selenium.scroll_element_into_view(btnlocator)
+      self.selenium.click_element(btnlocator)
+
+    @capture_screenshot_on_error
+    def select_value_from_rd2_modal_dropdown(self,dropdown,value):
+      """Selects given value in the dropdown field on the rd2 modal"""
+      locator = npsp_lex_locators["erd"]["modal_dropdown_selector"].format(dropdown)
+      selection_value = npsp_lex_locators["erd"]["modal_selection_value"].format(value)
+      if self.npsp.check_if_element_exists(locator):
+          self.selenium.set_focus_to_element(locator)
+          self.selenium.wait_until_element_is_visible(locator)
+          self.selenium.scroll_element_into_view(locator)
+          self.salesforce._jsclick(locator)
+          self.selenium.wait_until_element_is_visible(selection_value)
+          self.selenium.click_element(selection_value)
+
+    @capture_screenshot_on_error
+    def populate_rd2_modal_form(self, **kwargs):
+        """Populate the RD2 modal form fields with the respective fields and values"""
+        for key, value in kwargs.items():
+            if key in ("Amount", "Number of Planned Installments"):
+                locator = npsp_lex_locators["erd"]["modal_input_field"].format(key)
+                if self.npsp.check_if_element_exists(locator):
+                    self.selenium.set_focus_to_element(locator)
+                    self.salesforce._populate_field(locator, value)
+            if key in ("Account", "Contact"):
+                self.salesforce.populate_lookup_field(key, value)
+            else:
+                self.select_value_from_rd2_modal_dropdown(key,value)
 
 
 @pageobject("Details", "npe03__Recurring_Donation__c")
-class RDDetailPage(BaseNPSPPage,DetailPage ):
+class RDDetailPage(BaseNPSPPage, DetailPage):
     object_name = "npe03__Recurring_Donation__c"
     
+    
     def _is_current_page(self):
         """ Verify we are on the Account detail page
             by verifying that the url contains '/view'
         """
-        self.selenium.wait_until_location_contains("/view", timeout=60, message="Record view did not open in 1 min")
+        locator=npsp_lex_locators['bge']['button'].format("Edit")
+        edit_button=self.selenium.get_webelement(locator)
+        self.selenium.wait_until_page_contains_element(edit_button, error="Details page should not load fully")
```

Comment:
I think u mean error=" Details page did not load fully" please correct

Reply:
and first verify location and then verify the button otherwise it will verify the button on wrong page if it happen to be present on that page as well